src/compiler/commandLineParser.ts

```diff
@@ -3217,6 +3183,17 @@ namespace ts {
         return matchesExcludeWorker(pathToCheck, validatedExcludeSpecs, useCaseSensitiveFileNames, currentDirectory, basePath);
     }
 
+    function invalidDotDotAfterRecursiveWildcard(s: string) {
+        // We used to use the regex /(^|\/)\*\*\/(.*\/)?\.\.($|\/)/ to check for this case, but
+        // in v8, that has polynomial performance because the recursive wildcard match - **/ -
+        // can be matched in many arbitrary positions when multiple are present, resulting
+        // in bad backtracking (and we don't care which is matched - just that some /.. segment
+        // comes after some **/ segment).
+        const wildcardIndex = startsWith(s, "**/") ? 0 : s.indexOf("/**/");
+        const lastDotIndex = endsWith(s, "/..") ? s.length : s.lastIndexOf("/../");

```

Comment:
Does it matter that this won't catch leading `..`?

Reply:
Leading `..` can't come after a `**` segment, so.... nope.